pkg/proxy/userspace/proxysocket.go

```diff
@@ -73,7 +73,7 @@ func newProxySocket(protocol api.Protocol, ip net.IP, port int) (proxySocket, er
 }
 
 // How long we wait for a connection to a backend in seconds
-var endpointDialTimeout = []time.Duration{1, 2, 4, 8}
+var endpointDialTimeout = []time.Duration{250 * time.Millisecond, 500 * time.Millisecond, 2 * time.Second, 4 * time.Second}
```

Comment:
I'm not sure I understand why that is important?

On Thu, Oct 15, 2015 at 5:43 PM, Abhi Shah <notifications@github.com> wrote:

> In pkg/proxy/userspace/proxysocket.go
> <https://github.com/kubernetes/kubernetes/pull/15745#discussion_r42198977>
> :
>
> > @@ -73,7 +73,7 @@ func newProxySocket(protocol api.Protocol, ip net.IP, port int) (proxySocket, er
> >  }
> >
> >  // How long we wait for a connection to a backend in seconds
> > -var endpointDialTimeout = []time.Duration{1, 2, 4, 8}
> > +var endpointDialTimeout = []time.Duration{250 * time.Millisecond, 500 * time.Millisecond, 2 * time.Second, 4 * time.Second}
>
> I wanted to have a value higher than 2s to cover the 2s rtimeout on skydns.
> We can do 250ms, 512ms, 1s, 3s ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/15745/files#r42198977>.
>